src/main/java/com/android/tools/r8/ir/conversion/DexSourceCode.java

```diff
@@ -171,9 +176,77 @@ public void buildInstruction(
+   */
+  private void insertMissingExceptionPopIfNeeded(IRBuilder builder) {
+    // The logic checks here are a bit backwards, but this is for optimization.
+    // Generally what we want to do here is check if the current instruction offset
+    // marks the beginning of a catch block.
+    //
+    // If this is the case and the current instruction is moving the exception into
+    // a slot, when we know the exception will be used in the code, and we're done here.
+    //
+    // However, some people don't use the exception in the catch block. But we need to
+    // indicate in the IR that this value is on the stack and needs to be removed somehow.
+    // The easiest way to do that is to move it into register 0, which should be free
+    // because a catch block is entered after the stack is totally cleared.
+    if (currentDexInstruction.getClass() == DexMoveException.class) {
+      // We handle the exception, and as mentioned its assumed its being used.
+      // Nothing more to do here.
+      return;
+    }
+
+    // Doesn't seem like its being handled. We got to fix that if this is a catch block start.
+    // Let's detect that by tracking what exception types are handled at the current offset.
+    int currentInstructionOffset = currentDexInstruction.getOffset();
+    TryHandler[] handlers = code.getHandlers();
+    Set<DexType> typesAtCurrentOffset = new HashSet<>(2);
+    for (TryHandler handler : handlers) {
+      for (TypeAddrPair pair : handler.pairs) {
+        if (pair.addr == currentInstructionOffset) {
+          typesAtCurrentOffset.add(pair.getType());
+        }
+      }
+    }
+
+    // If we have more than one type we need to insert our handling logic.
+    int handledTypesAtCurrentOffset = typesAtCurrentOffset.size();
+    if (handledTypesAtCurrentOffset > 0) {
+        /*
+        DexType exceptionType;
+        if (handledTypesAtCurrentOffset == 1) {
+            exceptionType = typesAtCurrentOffset.iterator().next();
+        } else {
+          // TODO: Compute common type of exceptions, flattening to throwable is wrong
+          exceptionType = factory.throwableType;
+        }
+        ReferenceTypeElement exceptionTypeElement = factory.createReferenceTypeElement(exceptionType, Nullability.definitelyNotNull(), null);
+        builder.readRegister()
+        Value  exceptionValue = new Value(builder.getValueNumberGenerator().next(), exceptionTypeElement, null);
+        builder.add(new Pop(exceptionValue));
+        */
+
+      // Move into dummy first register. Should act like just like a POP.
+      // Again, this is fine because the exception isn't used.
+      // If it were used, the instruction here would be a 'DexMoveException' which we checked for earlier.
+
+
+      Set<Phi> latestBlockPhis = builder.getLatestBlockPhis();
+      DexMoveException dexMoveException = new DexMoveException(0);
+      dexMoveException.buildIR(builder);
+      builder.resetLatestBlockPhis(latestBlockPhis);
```

Comment:
The resetting here is a complete hack to see what happens if we ignore the complaint about phi's from the `MoveException` not being addressed properly.

I have no idea how to address them properly, but I'm hopeful that once that is figured out, this solution will work.

Reply:
Removed and replaced with a better fix